src/hardware/adc.rs

```diff
@@ -29,44 +51,66 @@ pub enum InputChannel {
     Seven = 7,
 }
 
+// Physical ADC devices on Thermostat
+#[derive(Clone, Copy)]
+
+pub enum AdcPhy {
+    Zero = 0,
+    One = 1,
+    Two = 2,
+    Three = 3,
+}
+
 type O = Output<PushPull>;
-type Adcs = (
-    ad7172::Ad7172<SharedBus<Spi<SPI4, Enabled>>, PE0<O>>,
-    ad7172::Ad7172<SharedBus<Spi<SPI4, Enabled>>, PE1<O>>,
-    ad7172::Ad7172<SharedBus<Spi<SPI4, Enabled>>, PE3<O>>,
-    ad7172::Ad7172<SharedBus<Spi<SPI4, Enabled>>, PE4<O>>,
-);
+type Adcs = ad7172::Ad7172<Spi<SPI4, Enabled>>;
 type SpiPins = (
     PE2<Alternate<AF5>>,
     PE5<Alternate<AF5>>,
     PE6<Alternate<AF5>>,
 );
-
+/// All pins for all ADCs.
+/// * `spi` - Spi clk, miso, mosi (in this order).
+/// * `cs` - The four chip select pins.
+/// * `rdyn` - ADC rdyn input (this is the same ad spi dout).
+/// * `sync` - ADC sync pin (shared for all adc phys).
 pub struct AdcPins {
+    pub spi: SpiPins,
     pub cs: (PE0<O>, PE1<O>, PE3<O>, PE4<O>),
+    pub rdyn: PC11<Input<PullUp>>,
+    pub sync: PB11<O>,
 }
 
 pub struct Adc {
     pub adcs: Adcs,
+    pub rdyn: PC11<Input<PullUp>>,
+    pub sync: PB11<O>,
+    pub cs: (PE0<O>, PE1<O>, PE3<O>, PE4<O>),
+    pub current_position: usize, // Schedule position
```

Comment:
Better name and comment.

```suggestion
    pub schedule_index: usize, // Currently active index into SCHEDULE
```